app/src/services/submission.ts

```diff
@@ -227,13 +228,44 @@ const service = {
       }
     });
 
+    if (!params.includeDeleted) {
+      const softDeletedActivities = await prisma.activity.findMany({ where: { is_deleted: true } });
+
+      // Remove soft deleted submissions
+      if (softDeletedActivities.length)
+        result = result.filter((x) => !softDeletedActivities.some((y) => y.activity_id === x.activity_id));
+    }
```

Comment:
`includeDeleted` should have a default value so this can be ran in a single query